extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -362,41 +367,51 @@ export class ProjectsController {
 		return projName;
 	}
 
-	private async getExtractTarget(): Promise<ExtractTarget> {
-		let extractTarget: ExtractTarget;
+	private getExtractTargetStrings(inputTarget: any): mssql.ExtractTarget {
+		if (inputTarget) {
+			switch (inputTarget) {
+				case 'File': return mssql.ExtractTarget['file'];
+				case 'Flat': return mssql.ExtractTarget['flat'];
+				case 'ObjectType': return mssql.ExtractTarget['objectType'];
+				case 'Schema': return mssql.ExtractTarget['schema'];
+				case 'SchemaObjectType': return mssql.ExtractTarget['schemaObjectType'];
+				default: throw new Error(`Invalid input: ${inputTarget}`);
+			}
+		} else {
+			throw new Error(constants.extractTargetRequired);
+		}
+	}
+
+	private async getExtractTarget(): Promise<mssql.ExtractTarget> {
+		let extractTarget: mssql.ExtractTarget;
 
 		let extractTargetOptions: QuickPickItem[] = [];
 
 		let keys: string[] = Object.keys(ExtractTarget).filter(k => typeof ExtractTarget[k as any] === 'number');
 
 		keys.forEach((targetOption: string) => {
-			if (targetOption !== 'dacpac') {
+			if (targetOption !== 'dacpac') {		//Do not present the option to create Dacpac
 				let pascalCaseTargetOption: string = utils.toPascalCase(targetOption);	// for better readability
 				extractTargetOptions.push({ label: pascalCaseTargetOption });
```

Comment:
FYI - a lot better way of handling this is to create a class that extends QuickPick item which also stores the values (in this case the enum values). Then you don't have to do all that mapping back to the original value stuff.

example :
https://github.com/microsoft/azuredatastudio/blob/master/extensions/azurecore/src/azureResource/commands.ts#L193

Reply:
Added a TODO for this.